webapp/channels/src/components/profile_popover/profile_popover_custom_attributes.tsx

```diff
@@ -25,12 +34,27 @@ const ProfilePopoverCustomAttributes = ({
             dispatch(getCustomProfileAttributeValues(userID));
         }
     });
+
     const attributeSections = customProfileAttributeFields.map((attribute) => {
-        if (userProfile.custom_profile_attributes) {
-            const value = userProfile.custom_profile_attributes[attribute.id];
-            if (!value) {
+        if (!hideStatus && userProfile.custom_profile_attributes) {
+            const visibility = attribute.attrs?.visibility || 'when-set';
+            if (visibility === 'never') {
                 return null;
             }
+
+            // Check if the attribute has a value
+            /* eslint-disable multiline-ternary */
+            const hasValue = userProfile.custom_profile_attributes[attribute.id] &&
+                (Array.isArray(userProfile.custom_profile_attributes[attribute.id]) ?
+                    userProfile.custom_profile_attributes[attribute.id].length > 0 :
+                    userProfile.custom_profile_attributes[attribute.id].length > 0);
+            /* eslint-enable multiline-ternary */
```

Comment:
Do we need this ternary, or would a null propagating work if the values are `string | string[]`?
```suggestion
            const hasValue = userProfile.custom_profile_attributes[attribute.id]?.length > 0;
```